tools/istio-iptables/pkg/builder/rules_builder.go

```diff
@@ -0,0 +1,52 @@
+package builder
+
+type Rule struct {
+	chain  *string
+	table  *string
+	params []string
+}
+
+type Rules struct {
+	rules   []*Rule
+	rulesv6 []*Rule
+}
+
+type Builder interface {
+	AddRule(chain *string, table *string, params ...string) Builder
+	AddRuleV6(chain *string, table *string, params ...string) Builder
+	Build() []*string
+	BuildV6() []*string
+	BuildIptablesRestoreFormat() *string
+	BuildIptablesV6RestoreFormat() *string
+}
+
+type RulesBuilder struct {
+}
+
+func New() *RulesBuilder {
```

Comment:
```suggestion
func NewIptablesRulesBuilder() *IptablesRulesBuilder {
```
^^^ I'm not very happy about `NewIptablesRulesBuilder`. It will stutter when called as `builder.NewIptablesRulesBuilder()`. How about:
```suggestion
func NewIptables() *IptablesRulesBuilder {
```

Reply:
yeah.. agree.. New() -> NewIptables